src/Internal/Payment/Change.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Class Change
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment;
+
+/**
+ * A change object used for logging.
+ * Key, former and current values are set as string.
+ */
+class Change {
+
+	/**
+	 * Identifier or key for the changed value.
+	 *
+	 * @var string
+	 */
+	private $key;
+
+	/**
+	 * Former value for the key.
+	 *
+	 * @var string|null
+	 */
+	private $old_value;
+
+	/**
+	 * Current value, updated aganist the key.
+	 *
+	 * @var string|null
+	 */
+	private $new_value;
+
+	/**
+	 * Constructs the class, with key, former and current values.
+	 *
+	 * @param string      $key Identifier or key for the changed value.
+	 * @param string|null $old_value Former value for the key.
+	 * @param string|null $new_value Current value, updated aganist the key.
+	 */
+	public function __construct( $key, $old_value, $new_value ) {
+		$this->key       = $key;
+		$this->old_value = $old_value;
+		$this->new_value = $new_value;
```

Comment:
Though this is not defined in the issue, but I think it's worth having timestamp here too @RadoslavGeorgiev ?

Reply:
Please see p1698094299150619/1697536048.736719-slack-C05997J5PCG (I'm about to share it to a wider audience soon). All async operations should happen within the (soon to be introduced) `Processing` state, and between it and the next state (ex. `Processed`). Considering we will be logging all state changes, we don't need timestamps for individual values.

I'll take a look at the PR changes now. We might need to add timestamps to transitions, but not individual var updates.

Reply:
Do you envisage transition becoming a more complex object? Currently, it is just a mapping of state names and the associated Changes.

Reply:
They would have a timestamp, but not much more.